diff --git a/lib/request.js b/lib/request.js
index 0fb63d6a..569f5ca8 100644
--- a/lib/request.js
+++ b/lib/request.js
@@ -248,19 +248,13 @@ Object.defineProperties(Request.prototype, {
   },
   port: {
     get () {
-      // first try taking port from host
-      const portFromHost = parseInt(this.host.split(':').slice(-1)[0])
-      if (!isNaN(portFromHost)) {
-        return portFromHost
+      const portReg = /(?<port>:\d+)$/
+      const host = this.headers.host ?? this.headers[':authority'] ?? ''
+      const matches = portReg.exec(host)
+      if (matches === null || matches[1] === undefined) {
+        return null
       }
-      // now fall back to port from host/:authority header
-      const host = (this.headers.host ?? this.headers[':authority'] ?? '')
-      const portFromHeader = parseInt(host.split(':').slice(-1)[0])
-      if (!isNaN(portFromHeader)) {
-        return portFromHeader
-      }
-      // fall back to null
-      return null
+      return parseInt(matches.groups.port.slice(1), 10)
     }
   },
   protocol: {
