diff --git a/lib/validation.js b/lib/validation.js
index a7ac2aea..d94b5ac8 100644
--- a/lib/validation.js
+++ b/lib/validation.js
@@ -269,7 +269,7 @@ function wrapValidationError (result, dataVar, schemaErrorFormatter) {
  */
 function getEssenceMediaType (header) {
   if (!header) return ''
-  return header.split(/[ ;]/, 1)[0].trim().toLowerCase()
+  return header.trimStart().split(/[ ;]/, 1)[0].trim().toLowerCase()
 }
 
 module.exports = {
diff --git a/test/schema-validation.test.js b/test/schema-validation.test.js
index 86010a5b..5d8fcf35 100644
--- a/test/schema-validation.test.js
+++ b/test/schema-validation.test.js
@@ -1460,6 +1460,27 @@ test('Schema validation will not be bypass by different content type', async t =
   t.assert.strictEqual(found.status, 400)
   t.assert.strictEqual((await found.json()).code, 'FST_ERR_VALIDATION')
 
+  let injected = await fastify.inject({
+    method: 'POST',
+    url: '/',
+    headers: {
+      'content-type': ' application/json'
+    },
+    payload: JSON.stringify({ invalid: 'string' })
+  })
+  t.assert.strictEqual(injected.statusCode, 400)
+  t.assert.strictEqual(injected.json().code, 'FST_ERR_VALIDATION')
+
+  injected = await fastify.inject({
+    method: 'POST',
+    url: '/',
+    headers: {
+      'content-type': ' application/json'
+    },
+    payload: JSON.stringify({ foo: 'string' })
+  })
+  t.assert.strictEqual(injected.statusCode, 200)
+
   found = await fetch(address, {
     method: 'POST',
     url: '/',
