diff --git a/lib/schema-controller.js b/lib/schema-controller.js
index 119c95c7..273ed5a4 100644
--- a/lib/schema-controller.js
+++ b/lib/schema-controller.js
@@ -1,8 +1,6 @@
 'use strict'
 
 const { buildSchemas } = require('./schemas')
-const SerializerSelector = require('@fastify/fast-json-stringify-compiler')
-const ValidatorSelector = require('@fastify/ajv-compiler')
 
 /**
  * Called at every fastify context that is being created.
@@ -21,9 +19,11 @@ function buildSchemaController (parentSchemaCtrl, opts) {
   }, opts?.compilersFactory)
 
   if (!compilersFactory.buildValidator) {
+    const ValidatorSelector = require('@fastify/ajv-compiler')
     compilersFactory.buildValidator = ValidatorSelector()
   }
   if (!compilersFactory.buildSerializer) {
+    const SerializerSelector = require('@fastify/fast-json-stringify-compiler')
     compilersFactory.buildSerializer = SerializerSelector()
   }
 
@@ -31,7 +31,7 @@ function buildSchemaController (parentSchemaCtrl, opts) {
     bucket: (opts && opts.bucket) || buildSchemas,
     compilersFactory,
     isCustomValidatorCompiler: typeof opts?.compilersFactory?.buildValidator === 'function',
-    isCustomSerializerCompiler: typeof opts?.compilersFactory?.buildValidator === 'function'
+    isCustomSerializerCompiler: typeof opts?.compilersFactory?.buildSerializer === 'function'
   }
 
   return new SchemaController(undefined, option)
