diff --git a/lib/schemas.js b/lib/schemas.js
index 02524c31..971c95c5 100644
--- a/lib/schemas.js
+++ b/lib/schemas.js
@@ -10,6 +10,7 @@ const {
   FST_ERR_SCH_DUPLICATE,
   FST_ERR_SCH_CONTENT_MISSING_SCHEMA
 } = require('./errors')
+const ContentType = require('./content-type')
 
 const SCHEMAS_SOURCE = ['params', 'body', 'querystring', 'query', 'headers']
 
@@ -147,10 +148,10 @@ function getSchemaSerializer (context, statusCode, contentType) {
     return false
   }
   if (responseSchemaDef[statusCode]) {
-    if (responseSchemaDef[statusCode].constructor === Object && contentType) {
-      const mediaName = contentType.split(';', 1)[0]
-      if (responseSchemaDef[statusCode][mediaName]) {
-        return responseSchemaDef[statusCode][mediaName]
+    const ct = new ContentType(contentType)
+    if (responseSchemaDef[statusCode].constructor === Object && ct.isValid) {
+      if (responseSchemaDef[statusCode][ct.mediaType]) {
+        return responseSchemaDef[statusCode][ct.mediaType]
       }
 
       // fallback to match all media-type
@@ -164,10 +165,10 @@ function getSchemaSerializer (context, statusCode, contentType) {
   }
   const fallbackStatusCode = (statusCode + '')[0] + 'xx'
   if (responseSchemaDef[fallbackStatusCode]) {
-    if (responseSchemaDef[fallbackStatusCode].constructor === Object && contentType) {
-      const mediaName = contentType.split(';', 1)[0]
-      if (responseSchemaDef[fallbackStatusCode][mediaName]) {
-        return responseSchemaDef[fallbackStatusCode][mediaName]
+    const ct = new ContentType(contentType)
+    if (responseSchemaDef[fallbackStatusCode].constructor === Object && ct.isValid) {
+      if (responseSchemaDef[fallbackStatusCode][ct.mediaType]) {
+        return responseSchemaDef[fallbackStatusCode][ct.mediaType]
       }
 
       // fallback to match all media-type
@@ -181,10 +182,10 @@ function getSchemaSerializer (context, statusCode, contentType) {
     return responseSchemaDef[fallbackStatusCode]
   }
   if (responseSchemaDef.default) {
-    if (responseSchemaDef.default.constructor === Object && contentType) {
-      const mediaName = contentType.split(';', 1)[0]
-      if (responseSchemaDef.default[mediaName]) {
-        return responseSchemaDef.default[mediaName]
+    const ct = new ContentType(contentType)
+    if (responseSchemaDef.default.constructor === Object && ct.isValid) {
+      if (responseSchemaDef.default[ct.mediaType]) {
+        return responseSchemaDef.default[ct.mediaType]
       }
 
       // fallback to match all media-type
