diff --git a/lib/validation.js b/lib/validation.js
index 21e0f165..ea76aee8 100644
--- a/lib/validation.js
+++ b/lib/validation.js
@@ -138,7 +138,7 @@ function validateParam (validatorFunction, request, paramName) {
   function answer (ret) {
     if (ret === false) return validatorFunction.errors
     if (ret && ret.error) return ret.error
-    if (ret && ret.value) request[paramName] = ret.value
+    if (ret && typeof ret === 'object' && 'value' in ret) request[paramName] = ret.value
     return false
   }
 }
