diff --git a/lib/decorate.js b/lib/decorate.js
index 6c725e92..966ef718 100644
--- a/lib/decorate.js
+++ b/lib/decorate.js
@@ -124,7 +124,7 @@ function checkDependencies (instance, name, deps) {
   }
 
   for (let i = 0; i !== deps.length; ++i) {
-    if (!checkExistence(instance, deps[i])) {
+    if (!checkExistence(instance, deps[i]) && !hasInstanceProperty(instance, deps[i])) {
       throw new FST_ERR_DEC_MISSING_DEPENDENCY(deps[i])
     }
   }
