diff --git a/src/rules.ts b/src/rules.ts
index cc2e62bc..0458ad28 100644
--- a/src/rules.ts
+++ b/src/rules.ts
@@ -275,7 +275,7 @@ const blockPedantic: Record<BlockKeys, RegExp> = {
 
 const escape = /^\\([!"#$%&'()*+,\-./:;<=>?@\[\]\\^_`{|}~])/;
 const inlineCode = /^(`+)([^`]|[^`][\s\S]*?[^`])\1(?!`)/;
-const br = /^( {2,}|\\)\n(?!\s*$)/;
+const br = /^( {2,}|\\)\n(?!\s*$)[ \t]*/;
 const inlineText = /^(`+|[^`])(?:(?= {2,}\n)|[\s\S]*?(?:(?=[\\<!\[`*_]|\b_|$)|[^ ](?= {2,}\n)))/;
 
 // list of unicode punctuation marks, plus any missing characters from CommonMark spec
